### PR TITLE
Issue #127

### LINKED ISSUE #127: restrict a user to report a particular DiscussionComment only one time
### Current Scenario

A user with `userId` can report a `DiscussionComment` with `commentId` multiple numbers of times. However, this should not be the case.

### Expected Scenario

Before creating a new document in the `ReportDiscussionCommentDetails` model, check if a document, having the same `userId` and `commentId` already exists. If this is the case, return `Comment already reported`; else create a new document in the `ReportDiscussionCommentDetails` model.

### PR DESCRIPTION
**Pull request by ankitvashisht12**
Fixes #127

### CODE REVIEW

File: src/models/ReportDiscussion.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    numberOfReports: {
+      type: Number,
+      required: true,
+    },
+  },
+  {
+    timestamps: true,
+  },
+);
+
```
Comment (Dishebh) [inline src/models/ReportDiscussion.js:19]: I guess this has been already implemented.

File: src/controllers/v1/discussion.js [hunk 2]
```diff
@@ -134,20 +134,33 @@ module.exports = {
...
+            userId,
+            discussionId,
+            discussionCommentId,
+          },
+        );
+
+        const reportDiscussionCommentDetails = await newReportDiscussionCommentDetails.save();
+
+        res.json({ data: reportDiscussionCommentDetails });
+      }
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:163]:
```suggestion
      }
        const newReportDiscussionCommentDetails = new ReportDiscussionCommentDetails(
          {
            reason,
            userId,
            discussionId,
            discussionCommentId,
          },
        );

        const reportDiscussionCommentDetails = await newReportDiscussionCommentDetails.save();

        return res.json({ data: reportDiscussionCommentDetails });
```

File: src/controllers/v1/discussion.js [hunk 3]
```diff
@@ -134,20 +134,33 @@ module.exports = {
...
           userId,
+          discussionCommentId,
         },
       );
 
-      const reportDiscussionCommentDetails = await newReportDiscussionCommentDetails.save();
-
-      res.json({ data: reportDiscussionCommentDetails });
+      if (hasreportDiscussionCommentDetails) {
+        res.json({ message: 'Comment already reported' });
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:149]:
```suggestion
        return res.status(400).json({ message: 'Comment already reported' });
```

File: src/controllers/v1/discussion.js [hunk 4]
```diff
@@ -134,20 +134,33 @@ module.exports = {
   reportDiscussionCommentDetails: create(
     async (req, res) => {
       const { reason } = req.body;
+      const { discussionId } = req.body;
       const discussionCommentId = req.params.commentId;
       const userId = req.user.id;
 
-      const newReportDiscussionCommentDetails = new ReportDiscussionCommentDetails(
+      const hasreportDiscussionCommentDetails = await ReportDiscussionCommentDetails.findOne(
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:141]:
```suggestion
      const hasReportDiscussionCommentDetails = await ReportDiscussionCommentDetails.findOne(
```

File: src/controllers/v1/discussion.js [hunk 5]
```diff
@@ -134,20 +134,33 @@ module.exports = {
   reportDiscussionCommentDetails: create(
     async (req, res) => {
       const { reason } = req.body;
+      const { discussionId } = req.body;
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:137]:
```suggestion
      const { reason, discussionId  } = req.body;
```

File: src/models/ReportDiscussion.js (hunk 1 above)
Comment (ankitvashisht12) [reply src/models/ReportDiscussion.js:19]: yeah ! i'll remove that

File: src/controllers/v1/discussion.js (hunk 2 above)
Comment (ankitvashisht12) [reply src/controllers/v1/discussion.js:163]: sure, i'll do that

File: src/controllers/v1/discussion.js (hunk 3 above)
Comment (ankitvashisht12) [reply src/controllers/v1/discussion.js:149]: on it

File: src/controllers/v1/discussion.js [hunk 6]
```diff
@@ -133,21 +133,34 @@ module.exports = {
 
   reportDiscussionCommentDetails: create(
     async (req, res) => {
-      const { reason } = req.body;
+      const { reason, discussionId } = req.body;
+
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:137]:
```suggestion
```

File: src/controllers/v1/discussion.js [hunk 7]
```diff
@@ -133,21 +133,34 @@ module.exports = {
...
       const discussionCommentId = req.params.commentId;
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:138]:
```suggestion
      const { reason } = req.body;
      const { discussionId, commentId } = req.params;
```

File: src/controllers/v1/discussion.js [hunk 8]
```diff
@@ -133,21 +133,34 @@ module.exports = {
...
       const userId = req.user.id;
 
+      const hasReportDiscussionCommentDetails = await ReportDiscussionCommentDetails.findOne(
+        {
+          userId,
+          discussionCommentId,
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:144]:
```suggestion
          discussionCommentId: commentId,
```

File: src/controllers/v1/discussion.js [hunk 9]
```diff
@@ -133,21 +133,34 @@ module.exports = {
...
+        return res.status(400).json({ message: 'Comment already reported' });
+      }
+
       const newReportDiscussionCommentDetails = new ReportDiscussionCommentDetails(
         {
           reason,
-          discussionCommentId,
           userId,
+          discussionId,
+          discussionCommentId,
```
Comment (Dishebh) [inline src/controllers/v1/discussion.js:157]:
```suggestion
          discussionCommentId: commentId,
```

**Pull request merged**